### PR TITLE
minetest: 5.9.1 -> 5.10.0, rename to luanti

### LINKED ISSUE #350344: minetest: Minetest has begun the process of changing to Luanti
A few days ago the Minetest devs announced they are changing the name of the project to Luanti.  https://blog.minetest.net/2024/10/13/Introducing-Our-New-Name/  
No breaking changes are being made (thankfully) so this is more to point out that nixpkgs should probably start looking at the steps to deprecate the `minetest` package names and start pointing them to `luanti` or the like, as well as updating git references to point to the new org location and what not.

## maintainers
@PyroLagus 
@fpletz
@fgaz

### PR DESCRIPTION
**Pull request by 06kellyjac**
Closes #350344

- **minetest: 5.9.1 -> 5.10.0**
- **minetest: rename to luanti**

https://blog.minetest.net/2024/10/13/Introducing-Our-New-Name/

image
image

Notes:

I removed a substitution for `/bin/rm` as it had no matches anymore
`pagezero_size` can probably be removed, I don't see any matches
Moved to pkgs/by-name and usage of `darwin.apple_sdk.frameworks.OpenAL` for darwin might need tweaking, it doesn't fallback on `openal` but IDK if it ever needed to
Have aliases in nixpkgs for minetest, added to the descriptiont to also help matches
Program still works with `minetest` because of symlinks

TODO:

Probably add release notes
Check if it needs backporting to 24.11

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [X] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by 06kellyjac** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 355231`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>luanti</li>
    <li>luanticlient</li>
    <li>luantiserver</li>
  </ul>
</details>

File: pkgs/by-name/lu/luanti/package.nix
Comment (06kellyjac) [inline pkgs/by-name/lu/luanti/package.nix:145]: https://www.luanti.org/ (and https://www.luanti.net/) are in-progress so leave minetest.net for now

**Review by Mindavi — APPROVED** [review]
Diff looks good to me.

**Comment by 06kellyjac** [comment]
I rebased incase this PR was affected by the latest treewide `pkgs/by-name` migration but it wasn't
I took the opportunity to add a release note.

File: pkgs/by-name/lu/luanti/package.nix
Comment (fgaz) [reply pkgs/by-name/lu/luanti/package.nix:145]: The migration is done, let's change it to luanti.org

**Review by fgaz — CHANGES_REQUESTED** [review]
Thanks!

This is a breaking change, so let's wait for the 24.11 branch-off before merging.

While we wait, there is a minetest module that I think we should rename as well.

Also could you please reformat the file in a separate commit so the diff is easier to review?

**Comment by fpletz** [comment]
> This is a breaking change, so let's wait for the 24.11 branch-off before merging.

Since the aliases are in place I don't think we need to wait for the branch-off. I think we should backport the name change anyway. It's a pity this is hitting right during our release freeze but this doesn't touch anything else anyway, right?

**Comment by 06kellyjac** [comment]
ok, commits as they should be. New url. I also updated the description to match their website & how they frame it more as a voxel game engine rather than just a game.

I'm also interested in helping maintenance for luanti/minetest if you'll have me. If so should I add it as a separate commit? :slightly_smiling_face:

**Comment by fgaz** [comment]
Regarding compatibility, the rename of `bin/minetest` in particular is my biggest concern, but isn't the only breaking change: https://dev.minetest.net/Changelog#5.9.1_.E2.86.92_5.10.0

So personally I wouldn't backport it either, but I don't feel strongly about it if the other maintainers disagree.

> I'm also interested in helping maintenance for luanti/minetest if you'll have me. If so should I add it as a separate commit? 🙂

Feel free to :)

**Comment by fpletz** [comment]
Mentioning @RossComputerGuy and @wegank as 24.11 release managers for their input on the breaking change involved here.

File: pkgs/games/minetest/default.nix
Comment (fgaz) [inline pkgs/games/minetest/default.nix]: Looks like they changed it to relative `"rm"` (https://github.com/minetest/minetest/pull/15238). I think we still want to patch it

Comment (06kellyjac) [reply pkgs/games/minetest/default.nix]: Good catch. Do you think `"rm"` is a good enough target for substitute?

Comment (fgaz) [reply pkgs/games/minetest/default.nix]: I'd write

```
--replace-fail 'argv = {
			"rm"' 'argv = { "${coreutils}/bin/rm"'
```

or if that doesn't work I'd use a patch instead

**Comment by RossComputerGuy** [comment]
Result of `nixpkgs-review pr 355231` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>luanti</li>
    <li>luanticlient</li>
    <li>luantiserver</li>
  </ul>
</details>

File: pkgs/top-level/aliases.nix
Comment (RossComputerGuy) [inline pkgs/top-level/aliases.nix:776]: I think these should be replaced with throws. Take a look at `mod_dnssd` which is a few lines down for an example.

Comment (06kellyjac) [reply pkgs/top-level/aliases.nix:776]: Thanks. I'm tweaking them to throws and fixing the formatting complaint

Any thoughts from people on dropping the `_5` items completely vs adding a `luanticlient_5` alias?
Also what to do with `minetest-touch`.
And maybe while we're renaming stuff, `minetestclient` -> `luanti-client` would be nice IMO

cc @fpletz @fgaz

Comment (fgaz) [reply pkgs/top-level/aliases.nix:776]: > Any thoughts from people on dropping the `_5` items completely vs adding a `luanticlient_5` alias? Also what to do with minetest-touch.

I'd drop both

> And maybe while we're renaming stuff, minetestclient -> luanti-client would be nice IMO

On one hand the server executable is named minetestserver so it makes sense to follow the same pattern. On the other, [all other distro packages](https://repology.org/project/luanti/versions) are dashed.

I agree with renaming.

**Comment by nyabinary** [comment]
Any status on this :p

**Comment by fgaz** [comment]
@06kellyjac are you still planning to finish this? let us know if you want us to take the PR to the finish line instead

**Review by bachp — APPROVED** [review]

**Comment by nyabinary** [comment]
> @06kellyjac are you still planning to finish this? let us know if you want us to take the PR to the finish line instead

No response, So I think you should prob supersede this atp?

File: pkgs/top-level/aliases.nix
Comment (fgaz) [reply pkgs/top-level/aliases.nix]: I dropped the _5 aliases completely, but I did not make the others throw since the module still uses `pkgs.minetest`

**Review by fgaz — APPROVED** [review]
Thanks everyone!

I fixed the conflict, made the changes we agreed on, moved the changelog entry to 25.05, and moved the `rm` substitution to the patch itself.

I'll merge in a bit.

**Comment by 06kellyjac** [comment]
Sorry, been busy over Christmas. Thanks @fgaz

**Pull request merged**
